api/routes/outlier_detecting.go

```diff
@@ -21,16 +21,28 @@ import (
 	"github.com/pkg/errors"
 	"goji.io/v3/pat"
 
+	"github.com/uncharted-distil/distil-compute/model"
 	api "github.com/uncharted-distil/distil/api/model"
 	"github.com/uncharted-distil/distil/api/task"
 )
 
+const (
+	outlierVarName     = "_outlier"
+	outlierDisplayName = "Outlier"
+)
+
+// OutlierResult represents a outlier response for a variable.
+type OutlierResult struct {
+	OutlierSuccess bool   `json:"success"`
+	OutlierField   string `json:"outlier"`
+}
+
 // OutlierDetectionHandler generates a route handler that enables outlier detection
-// for either remote sensing or tabular data
+// for either remote sensing or tabular data.
+// Return the name of the variable if the detection has run successfully.
 func OutlierDetectionHandler(metaCtor api.MetadataStorageCtor) func(http.ResponseWriter, *http.Request) {
```

Comment:
Clustering and feature ranking both run as a single call, creating columns etc. as part of the task before returning.  Was there a specific reason why outliers have been set up with two distinct routes?

Reply:
_Clustering_ is [re-applying the changes at each refresh of the page where the status bar is loaded.](https://github.com/uncharted-distil/distil/pull/2223/files#diff-7799ca076839513ade3b74fdf77f02549db37b0c0dbd82970256ceb5cdc75146R82)
And _Variable Ranking_ [keeps the results in the store and refresh them at every fetch of training variables.](https://github.com/uncharted-distil/distil/pull/2223/files?file-filters%5B%5D=.go&file-filters%5B%5D=.ts#diff-9f14e1174edfb8427cdc67df894afcd7b19e4046f7f42d55be7da91e38ae43baR455)

Because the requirement for this issue was to display the _Outlier Detection_ as a variable, I thought it more straightforward to add it to the back-end datasets; instead of testing on each view or state of the route when to add the outlier results. Now, once the page load, the _Outlier Detection_ runs, and if it's correct, the user can *apply* its result, which adds it to the back-end datasets (SQL & ES.)